api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/QualityImprovement2020CohortQueries.java

```diff
@@ -2303,6 +2315,63 @@ public CohortDefinition getPatientsFromFichaClinicaDenominatorB(String key) {
     return sqlCohortDefinition;
   }
 
+  public CohortDefinition getPatientsFromFichaClinicaDenominatorB1EOrB2E(boolean b1e) {
+
+    SqlCohortDefinition sqlCohortDefinition = new SqlCohortDefinition();
+    sqlCohortDefinition.setName("Patients From Ficha Clinica");
+    sqlCohortDefinition.addParameter(new Parameter("startDate", "startDate", Date.class));
+    sqlCohortDefinition.addParameter(new Parameter("endDate", "endDate", Date.class));
+    sqlCohortDefinition.addParameter(new Parameter("location", "location", Location.class));
+    sqlCohortDefinition.addParameter(
+        new Parameter("revisionEndDate", "revisionEndDate", Date.class));
+
+    Map<String, Integer> map = new HashMap<>();
+    map.put("6", hivMetadata.getAdultoSeguimentoEncounterType().getEncounterTypeId());
+    map.put("21151", hivMetadata.getTherapeuticLineConcept().getConceptId());
+    map.put("21150", hivMetadata.getFirstLineConcept().getConceptId());
+    map.put("21148", hivMetadata.getSecondLineConcept().getConceptId());
+    map.put("856", hivMetadata.getHivViralLoadConcept().getConceptId());
+
+    StringBuilder query = new StringBuilder();
+    query.append(" SELECT p.patient_id ");
+    query.append(" FROM patient p ");
+    query.append("    INNER JOIN encounter e ");
+    query.append("        ON p.patient_id = e.patient_id ");
+    query.append("    INNER JOIN obs o ");
+    query.append("               ON o.encounter_id = e.encounter_id ");
+    query.append("    INNER JOIN  (");
+    query.append("                    SELECT p.patient_id, MAX(e.encounter_datetime) max_date ");
+    query.append("                    FROM patient p ");
+    query.append("                        INNER JOIN encounter e ");
+    query.append("                            ON p.patient_id = e.patient_id ");
+    query.append("                    WHERE ");
+    query.append("                        p.voided = 0 ");
+    query.append("                        AND e.voided = 0 ");
+    query.append("                        AND e.encounter_type = ${6} ");
+    query.append("                        AND e.location_id = :location ");
+    query.append("                        AND e.encounter_datetime = :revisionEndDate ");
```

Comment:
group by clause missing here because of the max function

Reply:
done